File: examples/ruby/database/sqlite.rb
require "identifold"
require "identifold/storage/sqlite"
require "sqlite3"

connection = SQLite3::Database.new(":memory:")
connection.results_as_hash = true
connection.execute_batch(File.read("integrations/sqlite/migrations/001_identifold.up.sql"))
adapter = Identifold::Storage::SqliteAdapter.new(connection)
request = Identifold::Storage::ReferenceReservation.new(
  machine_id: "01890f8c-7b2a-7cc3-98b0-112233445566",
  namespace: "order",
  reference: "ORD-0123-4567-89-P"
)
reserved = adapter.reserve(request)
mapping = adapter.resolve(request.reference, request.namespace)
puts({ reserved: reserved, mapping: mapping }.inspect)
connection.close

File: examples/python/database/sqlite.py
from __future__ import annotations

import asyncio
import sqlite3
from pathlib import Path

from identifold.storage import ReferenceReservation
from identifold.storage.sqlite import SqliteStorageAdapter


async def main() -> None:
    connection = sqlite3.connect(":memory:", isolation_level=None)
    connection.row_factory = sqlite3.Row
    try:
        root = Path(__file__).resolve().parents[3]
        connection.executescript(
            (root / "integrations/sqlite/migrations/001_identifold.up.sql").read_text()
        )
        adapter = SqliteStorageAdapter(connection)
        request = ReferenceReservation(
            "01890f8c-7b2a-7cc3-98b0-112233445566",
            "order",
            "ORD-0123-4567-89-P",
        )
        reserved = await adapter.reserve(request)
        mapping = await adapter.resolve(request.reference, request.namespace)
        print({"reserved": reserved, "mapping": mapping})
    finally:
        connection.close()


asyncio.run(main())
